dist/plugin.js

```diff
@@ -35,30 +35,38 @@
 
         $.fn[name] = function(option) {
             var args = Array.prototype.slice.call(arguments);
+            var isMethodCall = typeof option === 'string';
+            var returnValue;
 
-            return this.each(function() {
-                var $this = $(this);
-                var plugin = $this.data(name);
-                var isMethodCall = typeof option === 'string';
+            if (isMethodCall) {
+                this.each(function() {
+                    var $this = $(this);
+                    var plugin = $this.data(name);
 
-                // If plugin isn't initialized, we lazy-load initialize it. If it's
-                // already initialized, we can safely ignore the call.
-                if (!plugin) {
-                    if (isMethodCall) {
+                    // We don't allow method calls on plugins that aren't initialized
+                    if (!plugin) {
                         throw new Error('cannot call methods on "' + name + '" prior to initialization; attempted to call method "' + option + '"');
                     }
-                    $this.data(name, (plugin = new SubConstructor(this, option)));
-                }
 
-                // invoke a public method on plugin, and skip private methods
-                if (isMethodCall) {
+                    // Skip private and non-existent methods
                     if (option.charAt(0) === '_' || typeof plugin[option] !== 'function') {
                         throw new Error('no such method "' + option + '" for "' + name + '"');
                     }
 
-                    plugin[option].apply(plugin, args.length > 1 ? args.slice(1) : null);
-                }
-            });
+                    returnValue = plugin[option].apply(plugin, args.length > 1 ? args.slice(1) : null);
```

Comment:
So for this code path we are going to return `returnValue` for the last time we call this function?

I'm wondering if the following function signature would make sense:

1) `this.length === 1` --> return `returnValue`
2) `this.length > 1` --> return `this` (to keep chaining possible). Not sure if this needs to be a new set of selector objects or if we could just use `this`.